acceptanceTests/cypress/integration/tests/UiTest.js

```diff
@@ -193,58 +193,60 @@ function checkFilter(filterDetails) {
-                cy.get('.w-tc-editor').should('have.attr', 'style').and('include', Cypress.env('greenFilterColor'));
-                if (!applyByCtrlEnter)
-                    cy.get('[type="submit"]').click();
+            waitForFetch();
+            pauseStream();
 
-                waitForFetch50AndPause();
+            // only one entry in DOM after filtering, checking all checks on it
+            leftTextCheck(entryId, leftSidePath, leftSideExpectedText);
+            leftOnHoverCheck(entryId, leftSidePath, filter);
 
-                // only one entry in DOM after filtering, checking all checks on it
-                leftTextCheck(entryId, leftSidePath, leftSideExpectedText);
-                leftOnHoverCheck(entryId, leftSidePath, filter);
+            rightTextCheck(rightSidePath, rightSideExpectedText);
+            rightOnHoverCheck(rightSidePath, filter);
+            checkRightSideResponseBody();
+        });
 
-                rightTextCheck(rightSidePath, rightSideExpectedText);
-                rightOnHoverCheck(rightSidePath, filter);
-                checkRightSideResponseBody();
-            });
+        resizeToHugeMizu();
 
-            resizeToHugeMizu();
+        // checking only 'leftTextCheck' on all entries because the rest of the checks require more time
+        cy.get(`#list [id^=entry]`).each(elem => {
+            const element = elem[0];
+            let entryId = getEntryId(element.id);
+            leftTextCheck(entryId, leftSidePath, leftSideExpectedText);
+        });
 
-            // checking only 'leftTextCheck' on all entries because the rest of the checks require more time
-            cy.get(`#list [id^=entry]`).each(elem => {
-                const element = elem[0];
-                let entryId = getEntryId(element.id);
-                leftTextCheck(entryId, leftSidePath, leftSideExpectedText);
-            });
+        // making the other 3 checks on the first X entries (longer time for each check)
+        deeperCheck(leftSidePath, rightSidePath, filter, rightSideExpectedText, entriesForDeeperCheck);
 
-            // making the other 3 checks on the first X entries (longer time for each check)
-            deeperCheck(leftSidePath, rightSidePath, filter, rightSideExpectedText, entriesForDeeperCheck);
+        // reloading then waiting for the entries number to load
+        resizeToNormalMizu();
+        cy.reload();
+        waitForFetch();
+        pauseStream();
+    });
+}
 
-            // reloading then waiting for the entries number to load
-            resizeToNormalMizu();
-            cy.reload();
-            cy.get('#total-entries', {timeout: refreshWaitTimeout}).should('have.text', totalEntries);
-        })
+function waitForFetch() {
+    cy.get('#entries-length', {timeout: refreshWaitTimeout}).should((el) => {
+        expect(parseInt(el.text().trim(), 10)).to.be.greaterThan(20);
```

Comment:
where is this "20" comes from?
I guess it should be some consts or parameter we are passing to the function but I know that we are sending 50 messages so I'm not sure why we are waiting only for "more than 20.."

Reply:
There are multiple tests using this function and I'm not sure what tests expects how many records. 20 was my guesstimation.

If all tests are 50+, we can say it's 50. Since WebSocket fetches up to 50 records.

I'm not sure what parameter and how exactly we should pass to here for the expected `greaterThan` value.

Reply:
I'm almost sure it is always 50 messages.
but I'm not sure we are sure it always catches all entries so I wouldn't go to 50.
lets start with a working tests and change it if we see that this can be changed.